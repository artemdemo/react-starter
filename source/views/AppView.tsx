import React from 'react';
import Container from '../components/Container/Container';

export const AppView: React.FC = (props) => {
  return (
    <Container>
      {props.children}
    </Container>
  );
};
