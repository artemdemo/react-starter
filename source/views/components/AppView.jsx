import React from 'react';
import history from '../../history';
import Icon from '../../components/Icon/Icon';
import Container from '../../components/Container/Container';
import Button from '../../components/Button/Button';
import MainMenu from '../../components/MainMenu/MainMenu';

import('../../services/asyncService').then(s => s.default());

const AppView = (props) => {
    return (
        <Container>
            <MainMenu />
            <p>
                <Button
                    type='button'
                    onClick={() => {
                        history.push('/third');
                    }}
                >
                    <Icon name='link' />
                    &nbsp;
                    Open third page programmatically
                </Button>
            </p>

            <hr />

            {props.children}
        </Container>
    );
};

export default AppView;
