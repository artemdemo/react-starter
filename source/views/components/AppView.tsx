import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faLink} from '@fortawesome/free-solid-svg-icons';
import {useHistory} from 'react-router-dom';
import Container from '../../components/Container/Container';
import Button, {EButtonAppearance} from '../../components/Button/Button';
import MainMenu from '../../containers/MainMenu/MainMenu';

const AppView: React.FC = (props) => {
  let history = useHistory();

  const goToThirdView = () => {
    history.push('/third');
  };

  return (
    <Container>
      <MainMenu/>
      <p>
        <Button
          type='button'
          onClick={goToThirdView}
          appearance={EButtonAppearance.LIGHT}
        >
          <FontAwesomeIcon icon={faLink}/>
          &nbsp;
          Open third page programmatically
        </Button>
      </p>

      <hr/>

      {props.children}
    </Container>
  );
};


export default AppView;
