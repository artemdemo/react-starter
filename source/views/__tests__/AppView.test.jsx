import React from 'react';
import renderer from 'react-test-renderer';
import AppView from '../components/AppView';

jest.mock('../../components/Icon/Icon');
jest.mock('../../components/Button/Button');
jest.mock('../../containers/MainMenu/MainMenu');
jest.mock('../../services/asyncService');
jest.mock('../../history');

describe('<AppView>', () => {
    const asyncServiceMock = require('../../services/asyncService');
    const historyMock = require('../../history');

    it('Simple render', () => {
        const tree = renderer.create(
            <AppView />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
        expect(asyncServiceMock.default).toBeCalledWith();
    });

    it('Render with children', () => {
        const tree = renderer.create(
            <AppView>
                Children
            </AppView>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should redirect to `/third`', () => {
        AppView.goToThirdView();

        expect(historyMock.default.push).toBeCalledWith('/third');
    });
});
