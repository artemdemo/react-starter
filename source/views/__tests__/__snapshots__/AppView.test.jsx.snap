// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppView> Render with children 1`] = `
<div
  className="container"
>
  <div
    data-mock="MainMenu"
  />
  <p>
    <div
      data-mock="Button"
    >
      {
  "type": "button",
  "appearance": "light"
}
      <div
        data-mock="Icon"
      >
        {
  "name": "link"
}
      </div>
        Open third page programmatically
    </div>
  </p>
  <hr />
  Children
</div>
`;

exports[`<AppView> Simple render 1`] = `
<div
  className="container"
>
  <div
    data-mock="MainMenu"
  />
  <p>
    <div
      data-mock="Button"
    >
      {
  "type": "button",
  "appearance": "light"
}
      <div
        data-mock="Icon"
      >
        {
  "name": "link"
}
      </div>
        Open third page programmatically
    </div>
  </p>
  <hr />
</div>
`;
